### PR TITLE
Making Kappa ready for CMSSW92X

### PR DESCRIPTION
**Pull request by rfriese**
Dear all,

after all a few changes had to be made to get Kappa Running on CMSSW. This is especially due to the TriggerObjectStandalone Collection that had some changes. They should be backward-compatible, but I would leave that to the automated tests complaining.

MVAMET is deactivated since there's no CMSSW9 training yet.

Also, the re-correction of jets is deactivated. As soon as there are relevant new JECs we should change that again. The reason is just that we used a module from the MVAMET implementation to re-correct the jets.

The refitted vertex does run. However if not totally necessary we should think about dropping it: Runtime-Wise it is the dominant module, by far: out of 0.166 s/event 0.15 s/event are due to the AdvancedRefitVertexBSProducer and AdvancedRefitVertexNoBSProducer producers on a 2017B SingleElectron Data file.

@greyxray is it a lot of work to extend the automated tests to the 92X release and run the old ones on this branch? I think nothing should change in the old releases, but this has to be proven.

### CODE REVIEW

**Comment by thomas-mueller** [comment]
Hallo Raphael,

currently CMSSW 8 is not covered by the automated travis tests (@greyxray can give more details), but maybe still by Fabio's tests. Only CMSSW 7 is covered by travis. So one might have to perform a manual test here...

I do not consider dropping the refitting as a good option since this would mean that Claudia would not be able to share Artus inputs with us but would need a complete own skim. We are more limited by storage space then by computing time.

Regards,
Thomas

**Comment by pistone87** [comment]
Hi Raphael,

I would like to keep the RefitVertex code, since it's crucial for my analysis.
If you want to drop something, you can drop the AdvancedRefitVertexBSProducer:
this is the refitting using the BS constraint and I actually don't need it anymore
(for the plots I need to do, I can use already produced old samples).
But the AdvancedRefitVertexNoBSProducer is pretty important for me.

Thanks.

Best,
Claudia

File: Skimming/higgsTauTau/kSkimming_run2_cfg.py [hunk 1]
```diff
@@ -145,7 +141,7 @@ def getBaseConfig( globaltag= 'START70_V7::All',
 	process.kappaTuple.active += cms.vstring('TriggerObjectStandalone')
 
 	# setup BadPFMuonFilter and BadChargedCandidateFilter
-	if tools.is_above_cmssw_version([8]):
+	if tools.is_above_cmssw_version([8]) and not tools.is_above_cmssw_version([8]):
```
Comment (anehrkor) [inline Skimming/higgsTauTau/kSkimming_run2_cfg.py]: Could it be that you forgot to change the latter version requirement to 9?

File: Skimming/higgsTauTau/kSkimming_run2_cfg.py [hunk 2]
```diff
@@ -541,12 +539,11 @@ def getBaseConfig( globaltag= 'START70_V7::All',
 
 	process.p *= process.nEventsFiltered
 	process.p *= process.nNegEventsFiltered
-	process.kappaTuple.active += cms.vstring('FilterSummary')
 
 	## ------------------------------------------------------------------------
 
 	## if needed adapt output filename
-	process.p *= process.kappaOut
+	process.ep *= process.kappaOut
```
Comment (anehrkor) [inline Skimming/higgsTauTau/kSkimming_run2_cfg.py:552]: Just for my curiosity: what is the difference between `process.p` and `process.ep`?

**Comment by anehrkor** [comment]
Just FYI: Olena is on holidays for at least a week. So you might have to wait some time for answers regarding the tests.

File: Skimming/higgsTauTau/kSkimming_run2_cfg.py (hunk 1 above)
Comment (rfriese) [reply Skimming/higgsTauTau/kSkimming_run2_cfg.py]: yes, thank you!

File: Skimming/higgsTauTau/kSkimming_run2_cfg.py (hunk 2 above)
Comment (rfriese) [reply Skimming/higgsTauTau/kSkimming_run2_cfg.py:552]: "p" is the path, "ep" is the endpath. Running Kappa in the path creates a unsolveable ordering problem of the unscheduled mode. By putting it to the endpath, everything is fine.

File: Skimming/higgsTauTau/kSkimming_run2_cfg.py [hunk 3]
```diff
@@ -163,29 +159,28 @@ def getBaseConfig( globaltag= 'START70_V7::All',
 			process.badGlobalMuonTaggerMAOD.taggingMode = cms.bool(True)
 			process.cloneGlobalMuonTaggerMAOD.taggingMode = cms.bool(True)
 
-			process.kappaTuple.TriggerObjectStandalone.metfilterbitslist = cms.vstring("BadChargedCandidateFilter", "BadPFMuonFilter", "badGlobalMuonTaggerMAOD", "cloneGlobalMuonTaggerMAOD")
-		else:
-			process.kappaTuple.TriggerObjectStandalone.metfilterbitslist = cms.vstring("BadChargedCandidateFilter","BadPFMuonFilter")
+#			process.kappaTuple.TriggerObjectStandalone.metfilterbitslist = cms.vstring("BadChargedCandidateFilter", "BadPFMuonFilter", "badGlobalMuonTaggerMAOD", "cloneGlobalMuonTaggerMAOD")
+#		else:
+#			process.kappaTuple.TriggerObjectStandalone.metfilterbitslist = cms.vstring("BadChargedCandidateFilter","BadPFMuonFilter")
```
Comment (anehrkor) [inline Skimming/higgsTauTau/kSkimming_run2_cfg.py]: Sorry for perhaps another stupid question, but do these lines need to be commented?

File: Skimming/higgsTauTau/kSkimming_run2_cfg.py (hunk 2 above)
Comment (anehrkor) [reply Skimming/higgsTauTau/kSkimming_run2_cfg.py:552]: I see. Thank you

**Pull request merged**